### PR TITLE
Migration: New datasource

### PR DESCRIPTION
**Pull request by tskarhed**
#22862

- Migrates datasource/new
- Adds possiblity to use `<Icon/>` for the `icon` props on Buttons. Any thoughts about this?

### CODE REVIEW

File: public/app/features/datasources/NewDataSourcePage.tsx [hunk 1]
```diff
@@ -151,17 +142,18 @@ const DataSourceTypeCard: FC<DataSourceTypeCardProps> = props => {
...
         {learnMoreLink && (
-          <a
-            className="btn btn-inverse"
+          <LinkButton
+            variant="secondary"
             href={`${learnMoreLink.url}?utm_source=grafana_add_ds`}
             target="_blank"
             rel="noopener"
             onClick={onLearnMoreClick}
+            icon={<Icon name="external-link" />}
```
Comment (tskarhed) [inline public/app/features/datasources/NewDataSourcePage.tsx]: Any thoughts about this usage?

Comment (torkelo) [reply public/app/features/datasources/NewDataSourcePage.tsx]: can't we just set icon to "external-link"

**Review by torkelo — APPROVED** [review]
Looks ok, some questions. Not sure why we can't pass icon as a string instead , and instead use React element

File: public/app/core/components/FilterInput/FilterInput.tsx [hunk 2]
```diff
@@ -1,24 +1,39 @@
...
   onChange: (value: string) => void;
+  useNewForms?: boolean;
 }
 
-export const FilterInput = forwardRef<HTMLInputElement, Props>((props, ref) => (
-  <label className={props.labelClassName}>
-    <input
-      ref={ref}
+export const FilterInput = forwardRef<HTMLInputElement, Props>((props, ref) =>
+  props.useNewForms ? (
```
[the remark this answers is not in the document]
Comment (tskarhed) [reply public/app/core/components/FilterInput/FilterInput.tsx:15]: Good point. I think the pages using it are fairly simple to migrate too, so we should probably be fine.

File: public/app/features/datasources/NewDataSourcePage.tsx (hunk 1 above)
Comment (tskarhed) [reply public/app/features/datasources/NewDataSourcePage.tsx:151]: It originally supported setting it as `fa fa-external-link` I think. Should maybe remove the `fa` then, and do some reasonable migration.

Comment (tskarhed) [reply public/app/features/datasources/NewDataSourcePage.tsx:151]: I'll use the fa string for now, and we'll see about that later.

**Pull request merged**